src/budget_transaction.rs

```diff
@@ -177,18 +185,24 @@ impl BudgetTransaction for Transaction {
         from_keypair: &Keypair,
         to: Pubkey,
         contract: Pubkey,
-        witness: Pubkey,
+        witness: &Vec<Pubkey>,
```

Comment:
I don't like this. Besides that clippy would tell you to use a slice, this interface doesn't reflect the "1 or 2" witnesses supported under the hood.

We shouldn't try to expose the `And` instruction in any general form. We can expose it in its very limited form, or do the work to change those `Payment` types to `Box<Budget>`, at which point, it could comfortably implement this interface.